### PR TITLE
Create the widget for date time question type

### LINKED ISSUE #173: Support DATE_TIME question type in the sdc library

### PR DESCRIPTION
**Pull request by jingtang10**
Closes #173

### CODE REVIEW

File: datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDateTimePickerViewHolderFactoryInstrumentedTest.kt [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            InstrumentationRegistry.getInstrumentation().targetContext,
+            R.style.Theme_MaterialComponents
+        )
+        parent = FrameLayout(context)
+        assertThat(parent).isNotNull()
+        viewHolder = QuestionnaireItemDateTimePickerViewHolderFactory.create(parent)
+    }
+
+    @Test
+    fun shouldSetTextViewText() {
```
Comment (joiskash) [inline datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDateTimePickerViewHolderFactoryInstrumentedTest.kt:56]: Should we rename this test to shouldSetTextInputLayoutHint in all the tests?

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDateTimePickerViewHolderFactory.kt [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+            private lateinit var timeInputLayout: TextInputLayout
+            private lateinit var timeInputEditText: TextInputEditText
+            private lateinit var questionnaireItemViewItem: QuestionnaireItemViewItem
+
+            override fun init(itemView: View) {
+                dateInputLayout = itemView.findViewById(R.id.dateInputLayout)
+                dateInputEditText = itemView.findViewById(R.id.dateInputEditText)
+                // Disable direct text input to only allow input from the date picker dialog
+                dateInputEditText.keyListener = null
+                dateInputEditText.setOnFocusChangeListener { _: View, hasFocus: Boolean ->
```
Comment (joiskash) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDateTimePickerViewHolderFactory.kt:50]: User experience : 
1. Open the date widget 
2. Select date
3. Press ok
4. Try to open the date widget again. It Does not open
If some one makes a mistake entering the date/time, the have to enter the time/date and then only then are they able to reopen the date widget. Not sure why this is happening. 

Also Looks like TextInputLayout remains highlighted if you click on a radio button / check box.

**Review by joiskash — CHANGES_REQUESTED** [review]

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDateTimePickerViewHolderFactory.kt (hunk 2 above)
Comment (jingtang10) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDateTimePickerViewHolderFactory.kt:50]: created #208

File: datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDateTimePickerViewHolderFactoryInstrumentedTest.kt (hunk 1 above)
Comment (jingtang10) [reply datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDateTimePickerViewHolderFactoryInstrumentedTest.kt]: done.

**Review by joiskash — APPROVED** [review]

**Pull request merged**